internal/provider/schema_gitlab_application_settings.go

```diff
@@ -0,0 +1,3054 @@
+			Type:        schema.TypeMap,
+			Elem:        &schema.Schema{Type: schema.TypeInt},
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"repository_storages": {
+			Description: "(GitLab 13.0 and earlier) List of names of enabled storage paths, taken from gitlab.yml. New projects are created in one of these stores, chosen at random.",
+			Type:        schema.TypeList,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"require_admin_approval_after_user_signup": {
+			Description: "When enabled, any user that signs up for an account using the registration form is placed under a Pending approval state and has to be explicitly approved by an administrator.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"require_two_factor_authentication": {
+			Description: "(If enabled, requires: two_factor_grace_period) Require all users to set up Two-factor authentication.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"restricted_visibility_levels": {
+			Description: "Selected levels cannot be used by non-Administrator users for groups, projects or snippets. Can take private, internal and public as a parameter. Default is null which means there is no restriction.",
+			Type:        schema.TypeList,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"rsa_key_restriction": {
+			Description: "The minimum allowed bit length of an uploaded RSA key. Default is 0 (no restriction). -1 disables RSA keys.",
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"send_user_confirmation_email": {
+			Description: "Send confirmation email on sign-up.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"session_expire_delay": {
+			Description: "Session duration in minutes. GitLab restart is required to apply changes.",
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"shared_runners_enabled": {
+			Description: "(If enabled, requires: shared_runners_text and shared_runners_minutes) Enable shared runners for new projects.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"shared_runners_minutes": {
+			Description: "Set the maximum number of CI/CD minutes that a group can use on shared runners per month.",
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"shared_runners_text": {
+			Description: "Shared runners text.",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"sidekiq_job_limiter_mode": {
+			Description: "track or compress. Sets the behavior for Sidekiq job size limits. Default: ‘compress’.",
```

Comment:
@timofurrer  - I think we should use the `valid values are: ` syntax here.

Reply:
Since this is generated I'd highly prefer to not do this unless someone really asks for this. Again, it would be too much effort to streamline them all ...